File: test/Github-Analyzer.Analysis.Tests/Fixtures/CSharp/Controllers/UserController.cs
using GithubAnalyzer.Fixtures.Services;
using GithubAnalyzer.Fixtures.Models;

namespace GithubAnalyzer.Fixtures.Controllers;

public class UserController
{
    private readonly UserService _userService;

    public UserController(UserService userService)
    {
        _userService = userService;
    }

    public User GetUser(int id)
    {
        return _userService.FindById(id);
    }

    public List<User> GetAllUsers()
    {
        return _userService.GetAll();
    }

    public void CreateUser(string name, string email)
    {
        var user = new User(name, email);
        _userService.Save(user);
    }
}

File: test/Github-Analyzer.Analysis.Tests/Fixtures/Php/Controllers/UserController.php
<?php

namespace App\Controllers;

require_once '../Services/UserService.php';

use App\Services\UserService;

class UserController
{
    private UserService $userService;

    public function __construct(UserService $userService)
    {
        $this->userService = $userService;
    }

    public function getUser(int $id): array
    {
        $user = $this->userService->findById($id);
        return $user;
    }

    public function getAllUsers(): array
    {
        return $this->userService->getAll();
    }

    public function createUser(string $name, string $email): void
    {
        $this->userService->save($name, $email);
    }

    public static function version(): string
    {
        return '1.0.0';
    }
}
